### ISSUE TITLE
Analytics Implementation or QA Support for VAMC

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#7336]
# Analytics Implementation or QA Request

## What this form is for
Use this template to request Google Tag Manager and Google Analytics implementation or QA for your product. Please fill out all brackets. 

> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints:
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

## Description
This request is for Google Analytics implementation 

Who is this request for?

    - Michelle Middaugh
    
What team/project is this work for?
    - Facilities: Left Nav enhancement 

Provide the Domain where Google Analytics needs to be implemented
   - https://www.va.gov/pittsburgh-health-care/
- There are no PII concerns for this effort. This is a publicly available web page,
    
Please provide any upcoming deadlines or important milestones
- Operating status metrics are a higher priority due to emerging COVID-19 needs. 
- The remaining tagging and metrics are lower priority but would be good to have on/near launch date: April 15 (tentative)
    
### Typical Website Interactions to Track

**Conversions - Understanding the User Flow:* 
- **Operating status:** 
- We need to track the number of conversions from clicking the COVID-19 alert from any page in VA.gov to the [Pittsburgh Operating status page](https://www.va.gov/pittsburgh-health-care/operating-status/) through to individual Facility-level operating status pages:
     - https://www.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/
     - https://www.va.gov/pittsburgh-health-care/locations/h-john-heinz-iii-department-of-veterans-affairs-medical-center/
     - https://www.va.gov/pittsburgh-health-care/locations/beaver-county-va-clinic/
     - https://www.va.gov/pittsburgh-health-care/locations/belmont-county-va-clinic/
     - https://www.va.gov/pittsburgh-health-care/locations/fayette-county-va-clinic/
     - https://www.va.gov/pittsburgh-health-care/locations/washington-county-va-clinic/
     - https://www.va.gov/pittsburgh-health-care/locations/westmoreland-county-va-clinic/

![covid alert](https://user-images.githubusercontent.com/55411834/77805942-e4fd8e80-7048-11ea-92c4-06d14bc78ad9.png)

![operating status](https://user-images.githubusercontent.com/55411834/77805943-e8911580-7048-11ea-8688-dada05be2fb4.png)

![facility notice](https://user-images.githubusercontent.com/55411834/77805956-f0e95080-7048-11ea-8909-4ab6e512287f.png)

![Facility operating status](https://user-images.githubusercontent.com/55411834/77805962-f3e44100-7048-11ea-9f7c-b9a544f23d2d.png)

*Other User Interactions*: 
1. **Metrics to assess user traffic on Left Nav** for Pittsburgh VAMC pages compared to "Top Tasks" and page content. The goal is to determine if the Left menu supports navigation to the most frequently visited page content. 
      - https://www.va.gov/pittsburgh-health-care/
      - https://www.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/
      - https://www.va.gov/pittsburgh-health-care/locations/h-john-heinz-iii-department-of-veterans-affairs-medical-center/
      - https://www.va.gov/pittsburgh-health-care/locations/beaver-county-va-clinic/
      - https://www.va.gov/pittsburgh-health-care/locations/belmont-county-va-clinic/
      - https://www.va.gov/pittsburgh-health-care/locations/fayette-county-va-clinic/
      - https://www.va.gov/pittsburgh-health-care/locations/washington-county-va-clinic/
      - https://www.va.gov/pittsburgh-health-care/locations/westmoreland-county-va-clinic/

![open state](https://user-images.githubusercontent.com/55411834/77806274-d82d6a80-7049-11ea-9b7e-fe289a356f65.png)

![top tasks](https://user-images.githubusercontent.com/55411834/77806312-f5facf80-7049-11ea-8d8a-209491084af2.png)

2. **# of clicks to reach content** for pages above

3. **Pogo-sticking navigation behavior**

> Please leave the following blank

## Acceptance Criteria
- [ ] QA has been completed
- [ ] VFS team completes brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed

**Comment by jonwehausen** [comment]
## Datalayer Requirements (Work in Progress) 

### READ ME - Developer Notes
- For clicks within the banner, the event in the table below can be applied to the other links within the alert box not shown in screenshot as well if needed
- Presently in production, we have an event for ```nav-accordion-expand``` or ```nav-accordion-collapse``` event for where the left nav is capable of being expanded or collapsed. **We would like to deprecate that convention in place of ```nav-sidenav-expand``` and ```nav-sidenav-collapse``` wherever present within the Pittsburgh health care sections.** Please find the new examples within the table below in rows 4+5
- The event ```nav-sidenav``` should continue to be used for such content that occurs **on a direct click of a left nav content item that is not expandable or collapseable and not within a child**
- The event ```nav-sidenav-child``` should be used for such nav components that are contained **within an expandable or collapseable nav component**. The variable ```'sidenav-dropdown-header'``` should populate with the dropdown header.


Interaction | Screenshot | Datalayer Requirements
------------ | -------------  | -------------
Clicks on links within the banner | ![image](https://user-images.githubusercontent.com/48527022/78252462-e1428f80-74c0-11ea-9ef2-864509dbeae3.png) | ```dataLayer.push({```<br>```'event':'nav-warning-alert-box-content-link-click',``` <br>```'alertBoxHeading': 'Help Limit the Spread of COVID-19 and other illnesses'```<br>```});```  | 
Clicks on Facilities in 'Facility operating statuses' list | ![image](https://user-images.githubusercontent.com/48527022/78252860-7f365a00-74c1-11ea-849a-75042f200476.png) | ```dataLayer.push({```<br>```'event':'nav-health-care-facility-click'});```
Clicks on Facility Notice Info Box | ![image](https://user-images.githubusercontent.com/48527022/78253647-b8bb9500-74c2-11ea-97a6-5da40613a021.png) | ```dataLayer.push({```<br>```'event': 'nav-info-box-click',```<br>```'infoBoxText': 'Facility Notice'});```
Clicks on Left Nav Dropdowns - Expands | ![image](https://user-images.githubusercontent.com/48527022/78256205-37fe9800-74c6-11ea-8373-300a0cc5b237.png) | ```dataLayer.push({```<br>```'event':'nav-sidenav-dropdown-expand',```<br>```'sidenav-dropdown-header': 'About us'});```
Clicks on Left Nav Dropdowns - Collapses | ![image](https://user-images.githubusercontent.com/48527022/78256916-29fd4700-74c7-11ea-9f80-b8667574001b.png) | ```dataLayer.push({```<br>```'event':'nav-sidenav-dropdown-collapse',```<br>```'sidenav-dropdown-header': 'About us'});```
Clicks on Left Nav - Child | ![image](https://user-images.githubusercontent.com/48527022/78258934-f8d24600-74c9-11ea-9695-94224b21f703.png)) | ```dataLayer.push({```<br>```'event':'nav-sidenav-child,```<br>```'sidenav-dropdown-header': 'Locations'});```
Clicks on Left Nav - Content Only | ![image](https://user-images.githubusercontent.com/48527022/78259925-5024e600-74cb-11ea-8274-5b1eda85548a.png) | ```dataLayer.push({```<br>```'event':'nav-sidenav'});```


### ✅ Existing Tracking For: 
- We currently have tracking already built for the jumplink clicks
![image](https://user-images.githubusercontent.com/48527022/78254107-6af35c80-74c3-11ea-842d-8e196847d91f.png)

**Comment by mmiddaugh** [comment]
Created https://github.com/department-of-veterans-affairs/va.gov-team/issues/7641 for datalayer on operating status

**Comment by jonwehausen** [comment]
## QA Testing of DL
tested at: http://35b73a51cb636ea4a3c5df5438a96ff3.review.vetsgov-internal/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/

Description of Interaction | Screenshot of DL | Status
------------ | ------------- | -------------
Click on "Get updates on affected services and facilities" | ![image](https://user-images.githubusercontent.com/48527022/79145259-569a4400-7d8e-11ea-9097-cd11a459587a.png) | ✅ **PASSED**
Click on "Pittsburgh VA Medical Center-University Drive" on operating status page | ![image](https://user-images.githubusercontent.com/48527022/79146181-f3111600-7d8f-11ea-8a43-a29ac4c5ae53.png) | ✅  **PASSED**
Click on "Facility operating status" jumplink | ![image](https://user-images.githubusercontent.com/48527022/79146744-d5907c00-7d90-11ea-90e0-db856c13479d.png) | ✅ **PASSED**

### Cannot Locate to Test
- @mmiddaugh we were not able to locate the "Facility Notice" button from the mock-up, could you give some insight into if this is still available? / Needs tracking? 
![image](https://user-images.githubusercontent.com/48527022/79147750-6e73c700-7d92-11ea-9303-7fee693f2822.png)

**Comment by bsmartin-ep** [comment]
@jonwehausen - 

Pt 1:
https://tagmanager.google.com/#/container/accounts/144676348/containers/2214120/workspaces/1000261

Pt 2:
https://tagmanager.google.com/#/container/accounts/144676348/containers/2214120/workspaces/1000269

**Comment by jonwehausen** [comment]
Still waiting on sidebar navigation events per note with Ethan this morning

**Comment by jonwehausen** [comment]
cc: @ethanteague @kelsonic @bsmartin-ep 

Per conversation with Ethan over Slack this morning, there was a design decision to steer away from the expandable / collapsible nav features and instead the "child" navigation components will automatically be exposed when the page renders. 

**Therefore, we no longer need events for `nav-sidenav-dropdown-expand` and `nav-sidenav-dropdown-collapse`**. 

We will now need to make sure the event `nav-sidenav-child` fires for those elements that are indented as a subheading and `nav-sidenav and test when available on staging.

**Comment by jonwehausen** [comment]
### QA Left Nav - Part 2

cc: @kelsonic @ethanteague @bsmartin-ep 

Description of interaction | DL Screenshot | Tag Screenshot | Test Status | Notes
------------ | ------------- | ------------- | -------------  | -------------
Click on a left nav item that is not indented | ![image](https://user-images.githubusercontent.com/48527022/81216051-426def00-8fa8-11ea-8603-965728ca7903.png) | ![image](https://user-images.githubusercontent.com/48527022/81216013-36822d00-8fa8-11ea-98ed-8bfc6f10888b.png) | ✔️ **PASSED** | n/a
Click on a left nav that is indented (child) | ![image](https://user-images.githubusercontent.com/48527022/81217673-fcfef100-8faa-11ea-9a35-21938e74844b.png)  | ![image](https://user-images.githubusercontent.com/48527022/81217264-4bf85680-8faa-11ea-9ebb-8ff5843c7bd4.png) | ✔️ **PASSED** | n/a

**Comment by jonwehausen** [comment]
## QA - Retest

Description of Interaction | DL Screenshot | Tag Screenshot | Test Status
------------ | ------------- | ------------- | -------------
Click on 'Locations' (parent of child) | ![image](https://user-images.githubusercontent.com/48527022/81300151-4ac83780-9045-11ea-839a-d6b766359012.png) | ![image](https://user-images.githubusercontent.com/48527022/81300407-9c70c200-9045-11ea-9fb5-6e0ac4c99719.png) | ✔️ **PASSED**

**Comment by jonwehausen** [comment]
Per conversation with @ethanteague over Slack, a production release date for the left nav and analytics events has not been set. Team is waiting final approval and expecting a release week of May 11 - 15, will follow up next week with determining hard deployment date.

**Comment by jonwehausen** [comment]
Spoke with Michelle today, left nav deployment is pushed back until 5/20

**Comment by jonwehausen** [comment]
Spoke with Etahn again today, VAMC is still working thru their staging review.

**Comment by jonwehausen** [comment]
Published and closed at 3pm EST today.
